nixops_encrypted_links/lib.py

```diff
@@ -124,16 +118,16 @@ def do_machine(m: nixops.backends.MachineState) -> None:
             m2 = active_machines[m2_name]
 
             # Don't create two tunnels between a pair of machines.
+            if not isinstance(m.index, int):
+                raise ValueError(f"{m.name} is missing an optional index required for encrypted-links")
+            if not isinstance(m2.index, int):
+                raise ValueError(f"{m2.name} is missing an optional index required for encrypted-links")
             if (
                 m.name
                 in to_encrypted_links_defn(self._machine_definition_for_required(m2.name)).encryptedLinksTo
                 and m.name >= m2.name
             ):
                 continue
-            if not m.index:
-                raise ValueError("Index was: {type(m.index)}")
-            if not m2.index:
-                raise ValueError("Index was: {type(m.index)}")
```

Comment:
These if statements were intended to check if the index of a machine, which is optional, exists.  Instead these statements were raising because the first machine in a cluster has index 0 which evals to true with a not and results in a raise.